### PR TITLE
Bl 972 problematic merge issues with auth 0 integration

### PR DESCRIPTION
**Pull request by tim-sansone**
## Description

We refactored several components to make use of axiosWithAuth directly. We decided that since application data is being stored directly in our database, we don't need to set that data into global state in the React app. So, we refactored several components to make the backend calls directly and deleted the files with the state-altering functions (which were being called by the components )entirely.

Fixes [BL-972](https://bloomtechlabs.atlassian.net/jira/software/c/projects/BL/boards/10?modal=detail&selectedIssue=BL-972)

[Loom Video](https://www.loom.com/share/d8d1a95eb2fc4a7fa1b4fb7faa08f144)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have removed unnecessary comments/console logs from my code
- [x] I have made corresponding changes to the documentation if necessary (optional)
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any misspellings
- [x] No duplicate code left within changed files
- [x] Size of pull request kept to a minimum
- [x] Pull request description clearly describes changes made & motivations for said changes

### CODE REVIEW

**Review by ezra-casas — APPROVED** [review]
Love it. Way to follow through and simplifying everything, this allows us to easily view the code and know what's happening, where things are being called from and where they're going.

File: src/components/pages/Navbar/Navbar.js [hunk 1]
```diff
@@ -82,21 +72,8 @@ const Navbar = ({ userProfile, getProfile, currentUser }) => {
...
-  // Determines the initial state of the Mentor toggle switch
   useEffect(() => {
-    dispatch(mentorInfo(profile_id))
-      .then(res => {
-        const availability = res.payload.data[0]?.availability;
-        setToggleStatus(availability);
-      })
-      .catch(err => console.log(err));
-    // eslint-disable-next-line react-hooks/exhaustive-deps
+    // We're leaving this useEffect here in case its needed later. User data that was previously loaded here will be developed in index.
```
Comment (mcfrr3) [inline src/components/pages/Navbar/Navbar.js:76]: Actually, now I'm noticing that there's a `useEffect` around line 54 🤦 . This `useEffect` can be safely removed

File: src/components/pages/PendingApplications/ApplicationModal.js [hunk 2]
```diff
@@ -93,26 +92,9 @@ const ApplicationModal = ({
...
-
-  /**
-   * Author: Khaleel Musleh
-   * @param {rejectApplication}
-   * Reject application dispatches a request to setApplicationReject in state/actions/applications which then returns a response of either a success or error status
-   */
-
-  const rejectApplication = status => {
-    dispatch(applicationReject(profileId, pendingAppHelper('rejected')))
+  const handleApplication = status => {
```
Comment (mcfrr3) [inline src/components/pages/PendingApplications/ApplicationModal.js:95]: Good job condensing this code. Always look for ways to make your code DRY

**Comment by mcfrr3** [comment]
Unless you're planning on doing more work with this, I would slap a video on there, link to the Jira ticket and open this up for review

**Review by ezra-casas — APPROVED** [review]
All Gucci.

File: src/components/pages/TicketsDashboard/TicketsDashboard.js [hunk 3]
```diff
@@ -1,5 +1,5 @@
 import React, { useEffect, useState } from 'react';
-import axiosWithAuth from '../../../utils/axiosWithAuth';
+import useAxiosWithAuth0 from '../../../hooks/useAxiosWithAuth0';
```
Comment (MaxT6) [inline src/components/pages/TicketsDashboard/TicketsDashboard.js:2]: Great job replacing axiosWithAuth here with the useAxiosWithAuth0 hook

File: src/components/pages/PendingApplications/PendingApplication.js [hunk 4]
```diff
@@ -97,65 +92,69 @@ const PendingApplications = ({ applicationProfile }) => {
...
   useEffect(() => {
     getPendingApps();
-    /**
-     * @Array {applicationProfile.length >= 0} Array
-     * @returns useEffect(() => {})
-     * Due to the dependency array not working on any state or variable due to the rendering being faster than data fetching which is passed through state
-     * I had to make a dependency array that renders once data length is 0 or higher.
-     */
-  }, [applicationProfile, dispatch]);
+  }, []);
```
Comment (MaxT6) [inline src/components/pages/PendingApplications/PendingApplication.js:157]: Nice job remembering the dependency array!

**Review by MaxT6 — APPROVED** [review]
Great job fixing the Auth 0 issues! It will be good to have the app functioning properly again.

**Comment by rberry16** [comment]
Looks good

**Review by rberry16 — APPROVED** [review]
Looks good

**Review by Nicholas-R-Payne — APPROVED** [review]
Looks good, great job fixing these issues with Auth0!

**Pull request merged**